docs/plugins/pluggable_rule_engine.md

```diff
@@ -32,29 +32,31 @@ Because the rule engine plugins are just one of many iRODS plugin types, the REP
 }
 ```
 
-Within the rule engine plugin framework, there are dynamically created policy enforcement points that are checked before and after every operation.  These are the "_pre", "_post", "_except", and "_finally" PEPs discussed in [Dynamic Policy Enforcement Points](dynamic_policy_enforcement_points.md).
+Within the REPF, there are dynamically created policy enforcement points that are checked before and after every operation.  These are the "_pre", "_post", "_except", and "_finally" PEPs discussed in [Dynamic Policy Enforcement Points](dynamic_policy_enforcement_points.md).
 
 The framework will look for rules that are defined with the same name as the PEP and execute them if found.  A typical `ils` will trigger over 1200 dynamic PEPs on a basic installation.  Nearly all of them will be undefined (there is no rule that matches their name) and so will not run any code.
 
 However, any that *do* match a particular PEP will be executed in the order in which they are loaded.  If there is only one matching rule, then it will fire and its return code will be interpreted by the REPF.  If it fails, then the operation fails as well and an error is returned to the client (or to the log when run by the [delay execution server](#delay-execution).)
 
-If there is more than one matching rule for a particular PEP, the first one loaded will fire first.  If it succeeds, then the others are ignored.  If the first one fails, then the next matching rule is fired.  If it fails, then the framework will continue to "fall through" until there are no more matching rules.  The return code of the last matching rule will be the one that is returned.
+If there is more than one matching rule for a particular PEP, only the first one loaded will fire.  If it succeeds, then the others are ignored.  If the first one fails, then an error is returned to the client and processing stops.
```

Comment:
I feel these lines can be combined in some way.

These statements can become false if a new rule language is implemented that supports its own form of rule lookup (i.e. fallthrough within the scope of a single REP).

Will have to think about the wording for this more.

Reply:
wait, is your edit correct?    within the entire framework, the first one failing *will* mean that the next defined PEP of the same name (in a different REP) will fire...   or am I still remembering before we squared up and introduced the RULE_ENGINE_CONTINUE logic...

Reply:
I think you may be thinking of the past behavior. Today, if a PEP fails, progress stops even if other PEP implementations exist.

Only `RULE_ENGINE_CONTINUE` allows PEP logic to be executed across REPs.

Reply:
Quite possible.

So you're saying that the earlier text *was* correct... or at least partially/mostly correct before the RULE_ENGINE_CONTINUE work we did... but the new behavior is what you've now described and are documenting.

Reply:
The original text may have been correct at the time when it was written. I can't make any claims about that. I also don't know when the behavior changed.

I believe I was able to demonstrate that the original text did not work for 4.2.2. That implies the change happened way before RULE_ENGINE_CONTINUE was introduced.

Reply:
okay, well, let's run with the new wordses.